Router and theme context

adriantd3 opened the pull request:
Two big things to consider here:

### Router

1.  It has been defined in a separate file so that we can keep the app.jsx file as clean as possible
2.  The "element" attribute defines the layout of the path, whereas the "children" array determines the content of the path. That is why every other route is a root's child.
3. That content will be placed inside the Outlet component. The RootLayout component contains it inside Layout.Content.
4. To navigate from pages, you must use the Link component from react router. An example can be found in the WikiHeader.

### ThemeContext

1.  The ThemeContext just configures the theme that will be used in the ConfigProvider. The ConfigProvider is a component created by Ant Design that lets you define some general config for its children.
2. For the themes, we are using the default themes by Ant Design
3. The ant design components will be styled following the theme that is set on the ConfigProvider. Moreover, those values can be used as css variables and also obtained with a hook
4. Could not find the css variables on the docs, but if you inspect elements, you will be able to see them all.

Check it out pls <3

adriantd3 committed:
Router created

adriantd3 committed:
ThemeContext created

Defined a context that provides the current theme and the function to toggle it.

It has been created using the Ant Design themes algorithm. The algorithm provides a way to derivate some secondary colors by defining some primary colors as a base.

We are using the light and dark default preset that Ant design provides.

You can use the css variables it generates on your css files

JoseBueno30 commented on front/antd-demo/src/layout/router.jsx:
```diff
@@ -0,0 +1,77 @@
+              path: ":wiki_id/articles",
+              children: [
+                {
+                  path: "search",
```
Delete this path. It's repeated (wiki_id/search) and could provoke conflicts if an article is called "search" because of articles/article_name

JoseBueno30 commented on front/antd-demo/src/layout/router.jsx:
```diff
@@ -0,0 +1,77 @@
+      path: "/",
+      element: <RootLayout />,
+      children: [
+        { path: "/", element: <TestRoute /> },
```
Add path that representes the search page for wikis

JoseBueno30 commented on front/antd-demo/src/layout/router.jsx:
```diff
@@ -0,0 +1,77 @@
+              element: <TestRoute />,
+            },
+            {
+              path: ":wiki_id/articles",
```
change <element>_id for <element>_name, the path should work with the name of the elements so it's more readeable by the user

adriantd3 committed:
Several fixes to router

- Changed the _id to _name
- Created a new route for creating articles and wikis
- Removed the duplicated route for searching articles

adriantd3 committed:
Merge branch 'main' into atd-router-init

adriantd3 committed:
Cleared the App.jsx file
